### PR TITLE
Add -function option and user specified arguments

### PR DESCRIPTION
**Pull request by arianneb**
Add option to take command line arguments for user specified function. I tried to remove the white space changes at the end of this commit for like 20 minutes. It's not happening. Sorry.

### CODE REVIEW

File: b9run/main.cpp [hunk 1]
```diff
@@ -86,14 +96,9 @@ static bool parseArguments(RunConfig& cfg, const int argc, char* argv[]) {
...
-  // positional
-
+  // check for user defined module
   if (i < argc) {
-    cfg.moduleName = argv[i++];
-
-    if (i < argc) {
-      cfg.mainFunction = argv[i++];
-    }
+    cfg.moduleName = argv[i];
```
Comment (rwy7) [inline b9run/main.cpp]: This should still be `argv[i++];`

**Comment by rwy7** [comment]
So far this is looking good. I think instead of `std::vector<int>`, you should use `std::vector<StackElement>` for usrArgs.

Don't forget to update the usage string as well 👍

**Review by xliang6 — DISMISSED** [review]
Can you change the commit title since this is a public repository? Sorry to nitpick, thanks.

File: b9/core.cpp [hunk 2]
```diff
@@ -352,24 +352,35 @@ void ExecutionContext::reset() {
...
   if (cfg_.verbose)
     std::cout << "function: " << functionIndex << " nargs: " << f->nargs
               << std::endl;
 
-  for (std::size_t i = 0; i < f->nargs; i++) {
-    int arg = 100 - (i * 10);
-    std::cout << "Pushing arg[" << i << "] = " << arg << std::endl;
-    executionContext_.push(arg);
+  if (usrArgs.size() == 0) {
+    // push random arguments to send to the program
```
Comment (rwy7) [inline b9/core.cpp]: don't push random args, just delete this branch,

Comment (arianneb) [reply b9/core.cpp]: @rwy0717 what if the user doesn't specify arguments for a function the requires them? error out?

Comment (rwy7) [reply b9/core.cpp]: Yep, throw an exception, and catch it in b9run.

**Pull request merged**